strawberryfields/apps/train/__init__.py

```diff
@@ -3,19 +3,87 @@
 
 .. currentmodule:: strawberryfields.apps.train
 
+A GBS device can be treated as a variational quantum device with trainable squeezing,
+beamsplitter and rotation gates. This module provides the functionality for training a GBS device
+using the approach outlined in `this <https://arxiv.org/abs/2004.04770>`__ paper
+:cite:`banchi2020training`.
+
+Embedding trainable parameters in GBS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Training algorithms for GBS distributions rely on the :math:`WAW` parametrization, where :math:`W`
+is a diagonal matrix of weights and :math:`A` is a symmetric matrix. Trainable parameters are
+embedded into the GBS distribution by expressing the weights as functions of the parameters.
+
+This module contains methods to implement such embeddings. It also provides derivatives
+of the weights with respect to the trainable parameters. There are two main classes, each
+corresponding to a different embedding. The :class:`Exp` class is a simple embedding where the
+weights are exponentials of the trainable parameters. The :class:`ExpFeatures` class is a more
+general embedding that makes use of user-defined feature vectors, which potentially provide more
+flexibility in training strategies.
```

Comment:
This is taken from the `embed` module.

Reply:
We may come back to change wording here if needed, but to keep this PR light I have just copy/paste (+ a minor change due to the move).

Reply:
I think it read well and could be left as is